source/unit/Constraints.ooc

```diff
@@ -131,6 +139,10 @@ EqualModifier: class extends Modifier {
 		f := func (value, c: Object) -> Bool { value as Cell<Double> get() == c as Cell<Double> get() }
 		CompareWithinConstraint new(this, Cell<Double> new(correct), f)
 	}
+	to: func ~ldouble (correct: LDouble) -> CompareWithinConstraint {
+		f := func (value, c: Object) -> Bool { value as Cell<LDouble> get() == c as Cell<LDouble> get() }
+		CompareWithinConstraint new(this, Cell<LDouble> new(correct), f)
+	}
```

Comment:
The duplicate code here makes me sad. Say, would this type of thing be avoidable in e.g. C++ by using e.g. templates?